### PR TITLE
Tweak CC fill

### PR DESCRIPTION
**Pull request by GioSensation**
**Reviewer:** @alistairjcbrown 
**Asana:** https://app.asana.com/0/0/1206101965623031/f

## Description
See the [Asana task](https://app.asana.com/0/0/1206101965623031/f) for more info. Please ignore the test form changes, it's just a reformat and removing broken markup with nested `form` elements. We added the nested `form`s to emulate the actual markup of using iframes. I've just changed it to divs. It's good enough for our needs in those specific tests.

## Steps to test
Tested in the PFR. More info in the task.

### CODE REVIEW

File: integration-test/tests/login-form.macos.spec.js [hunk 1]
```diff
@@ -348,18 +348,18 @@ test.describe('Auto-fill a login form on macOS', () => {
                     username: personalAddress,
                     password
                 },
-                expectedLabel: 'Manage Logins…'
+                expectedLabel: 'Manage passwords…'
```
Comment (GioSensation) [inline integration-test/tests/login-form.macos.spec.js:351]: Just changed this term and changed to lowercase.

File: src/Form/Form.js [hunk 2]
```diff
@@ -370,14 +370,22 @@ class Form {
...
-            if (foundInputs.length === 0 && this.form instanceof HTMLFormElement && this.form.length > 0) {
-                foundInputs = this.form.parentElement?.querySelectorAll(selector) || foundInputs
+            /** @type {Element[] | NodeList} */
+            let foundInputs = []
+
+            if (this.form instanceof HTMLFormElement) {
+                // For form elements we use .elements to catch fields outside the form itself using the form attribute.
+                // It also catches all elements when the markup is broken.
+                // We use .filter to avoid fieldset, button, textarea etc.
+                foundInputs = [...this.form.elements].filter(el => el.matches(selector))
```
Comment (GioSensation) [inline src/Form/Form.js:384]: The comment says it all, it's to ensure that fields outside the form tree, but with the `form` attribute are accounted for.

File: src/Form/Form.js [hunk 3]
```diff
@@ -761,26 +769,33 @@ class Form {
     autofillData (data, dataType) {
         this.isAutofilling = true
 
-        this.execOnInputs((input) => {
-            const inputSubtype = getInputSubtype(input)
-            let autofillData = data[inputSubtype]
+        const isActiveInputCCName = this.activeInput ? getInputSubtype(this.activeInput) === 'cardName' : false
+        // If autofill is initiated from a credit card name field, we only fill the name, not the entire form
```
Comment (GioSensation) [inline src/Form/Form.js:773]: This is the core change.

File: src/Form/test-cases/index.json [hunk 4]
```diff
@@ -105,7 +105,7 @@
   { "html": "chewy_signup.html" },
   { "html": "chewy_checkout.html", "expectedSubmitFalseNegatives": 2 },
   { "html": "everydayoil_checkout.html", "expectedSubmitFalseNegatives": 2, "expectedSubmitFalsePositives": 4 },
-  { "html": "stripeelements_checkout.html", "expectedSubmitFalseNegatives": 3 },
+  { "html": "stripeelements_checkout.html", "expectedSubmitFalseNegatives": 2 },
```
Comment (GioSensation) [inline src/Form/test-cases/index.json:108]: Positive side-effect.

**Review by alistairjcbrown — APPROVED** [review]
Code LGTM, fix works as expected on provided build :shipit:

**Pull request merged**